src/Components/Authentication/__tests__/Mobile/SignUpForm.test.tsx

```diff
@@ -16,14 +27,120 @@ describe("MobileSignUpForm", () => {
+    wrapper.find(SubmitButton).simulate("click")
+
+    setTimeout(() => {
+      wrapper.update()
+      const inputPass = wrapper.find(QuickInput).instance() as QuickInput
+      inputPass.onChange(ChangeEvents.password)
+      wrapper.find(SubmitButton).simulate("click")
+
+      setTimeout(() => {
+        wrapper.update()
+        expect(wrapper.find(QuickInput).props().label).toEqual("Name")
+        expect(wrapper.find(QuickInput).props().type).toEqual("text")
+        done()
+      })
+    })
+  })
+
+  describe("onSubmit", () => {
+    it("calls handleSubmit with expected params", done => {
+      const wrapper = getWrapper()
+      const inputEmail = wrapper.find(QuickInput).instance() as QuickInput
+      const termsCheckbox = wrapper.find(`Checkbox`).instance() as Checkbox
+      inputEmail.onChange(ChangeEvents.email)
+      termsCheckbox.onChange(ChangeEvents.accepted_terms_of_service)
+      wrapper.find(SubmitButton).simulate("click")
+
+      setTimeout(() => {
+        wrapper.update()
+        const inputPass = wrapper.find(QuickInput).instance() as QuickInput
+        inputPass.onChange(ChangeEvents.password)
+        wrapper.find(SubmitButton).simulate("click")
+
+        setTimeout(() => {
+          wrapper.update()
+          const inputName = wrapper.find(QuickInput).instance() as QuickInput
+          inputName.onChange(ChangeEvents.name)
+          wrapper.find(SubmitButton).simulate("click")
+
+          setTimeout(() => {
+            expect(props.handleSubmit.mock.calls[0][0]).toEqual({
+              email: "email@email.com",
+              accepted_terms_of_service: true,
+              password: "password",
+              name: "User Name",
+            })
+            done()
+          })
+        })
+      })
+    })
+
+    it("fires reCAPTCHA even", done => {
+      const wrapper = getWrapper()
+      const inputEmail = wrapper.find(QuickInput).instance() as QuickInput
+      const termsCheckbox = wrapper.find(`Checkbox`).instance() as Checkbox
+      inputEmail.onChange(ChangeEvents.email)
+      termsCheckbox.onChange(ChangeEvents.accepted_terms_of_service)
+      wrapper.find(SubmitButton).simulate("click")
+
+      setTimeout(() => {
+        wrapper.update()
+        const inputPass = wrapper.find(QuickInput).instance() as QuickInput
+        inputPass.onChange(ChangeEvents.password)
+        wrapper.find(SubmitButton).simulate("click")
+
+        setTimeout(() => {
+          wrapper.update()
+          const inputName = wrapper.find(QuickInput).instance() as QuickInput
+          inputName.onChange(ChangeEvents.name)
+          wrapper.find(SubmitButton).simulate("click")
+
+          setTimeout(() => {
+            expect(window.grecaptcha.execute).toBeCalledWith(
+              "recaptcha-api-key",
+              {
+                action: "signup_submit",
+              }
+            )
+            done()
+          })
+        })
+      })
+    })
```

Comment:
@eessex - have you explored Jest's async / await / promise features? (`resolves`, etc.) All of these setTimeouts seem like a bit of a flag, (though totally understand needing to use them from time to time).

Reply:
Yes I am familiar with the api but as stated above, the async behavior is not related to `recaptcha`. I'm wary to mock the actual form behavior because I don't know it very well and could not see an obvious place that the async originates, but did add a comment that this should be fixed.  As far as the tests I fixed the ones that failed, and extended the existing ones to make sure we got to the last step where the captcha is called.